src/projects.rs

```diff
@@ -67,6 +101,46 @@ fn fetch_next_item(config: &Config, project_name: &str) -> Result<Option<Item>,
     Ok(items.first().map(|item| item.to_owned()))
 }
 
+/// Fetch projects and prompt to add them to config one by one
+pub fn import(config: Config) -> Result<String, String> {
+    let projects = request::projects(&config)?;
+    let new_projects = filter_new_projects(&config, projects);
+    let mut config = config;
```

Comment:
we probably could get reed of this sort of reassignment by changing the signatures of some functions

but it is so minor that I don't know if it is beneficial to do at all. just bugs me a little =)

```rust
#[cfg(not(tarpaulin_include))]
fn project_import(matches: &ArgMatches) -> Result<String, String> {
    let mut config = fetch_config(matches)?;

    projects::import(&mut config)
}

//change import fn to receive mut ref
pub fn import(config: &mut Config) -> Result<String, String> {
    let projects = request::projects(config)?;
    let new_projects = filter_new_projects(config, projects);
    for project in new_projects {
        maybe_add_project(config, project)?;
    }
    Ok(green_string("No more projects"))
}
```

and change a few existing assignments 
note that way in import fn
```rust
    let projects = request::projects(config)?;
    let new_projects = filter_new_projects(config, projects);
```
we don't strictly need &config, our config is already &mut, and Rust compiler can infer it from the context and signatures of the functions to which this ref is being provided, and "transform our &mut to &"

Just find it interesting. We can opt-in and be explicit but clippy will complain about it by default

Reply:
Cool, thank you. I've been avoiding passing mutable references up until now but it's obviously the better way to do it in Rust. :+1: